src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -1186,8 +1186,6 @@ public abstract partial class JsonTypeInfo
         public System.Text.Json.Serialization.Metadata.JsonPolymorphismOptions? PolymorphismOptions { get { throw null; } set { } }
         public System.Collections.Generic.IList<System.Text.Json.Serialization.Metadata.JsonPropertyInfo> Properties { get { throw null; } }
         public System.Type Type { get { throw null; } }
-        [System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute("JSON serialization and deserialization might require types that cannot be statically analyzed and might need runtime code generation. Use System.Text.Json source generation for native AOT applications.")]
-        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("JSON serialization and deserialization might require types that cannot be statically analyzed and might need runtime code generation. Use System.Text.Json source generation for native AOT applications.")]
```

Comment:
Can't comment on the actual line, but could we also remove RUC from the generic `CreateJsonTypeInfo` method as well?

Reply:
I can try to double check after this PR but I think the answer is no because we need to get converter which by default requires RUC, I think we're able to do that once we get converter as an arg

Reply:
Is throwing if default converters have been rooted a viable option? Would exposing an overload in the future that accepts JsonConverter be a possible linker-safe alternative?